### PR TITLE
fix(gridintersect): clean up of gridintersect

### PR DESCRIPTION
**Pull request by dbrakenhoff**
This PR does the following:
- Add DeprecationWarning when `method != "vertex"`
- Remove all Shapely<2.0 code
- Remove warnings filters for older numpy versions
- Mark methods/tests that should be removed when `method="structured"` is officially removed
- Ensure full test suite is maintained after removal of structured tests.
- Move raster tests to a separate file.
- Add option to return intersection results as (Geo)DataFrame
- Allow plotting options to take as input GeoDataFrame (bit unnecessary as they have their own plotting logic) and improve plotting results.
- Add plot_intersection_result() to plot result + modelgrid (optional). 
- Update example notebook.

Note:
- Tests for 3D point intersections above/inside grid are deactivated. This is not yet working for `method="vertex"`. Will probably be added in a separate PR. 

I'm not sure what the next release will be so I marked most things deprecated (with warning) in the next release and to be removed by 3.10.0. Let me know if that makes sense. 

EDIT: modify removal version to 3.10.0

### CODE REVIEW

File: flopy/utils/gridintersect.py [hunk 1]
```diff
@@ -212,7 +169,7 @@ def intersect(
         return_all_intersections=False,
         contains_centroid=False,
         min_area_fraction=None,
-        shapely2=True,
```
Comment (wpbonelli) [inline flopy/utils/gridintersect.py:215]: Maybe worth keeping the `shapely2` parameter for now to avoid breakage, and warn if provided?

**Review by wpbonelli — CHANGES_REQUESTED** [review]
I think the next release will probably be 3.9 to coincide with mf6.6 next month? Per the [documented deprecation policy](https://github.com/modflowpy/flopy/blob/develop/DEVELOPER.md#deprecation-policy) we should then wait 2 minor releases before removing but IMO deprecating in 3.9 and dropping in 3.10 is fine &mdash; minor releases may slow down as development effort shifts toward the next major version

File: flopy/utils/gridintersect.py (hunk 1 above)
Comment (dbrakenhoff) [reply flopy/utils/gridintersect.py:215]: Done.

**Comment by dbrakenhoff** [comment]
There will probably be a merge conflict for this PR once #2343 is merged, I'll fix it when that happens.

**Review by wpbonelli — APPROVED** [review]

**Review by langevin-usgs — APPROVED** [review]
Nice cleanup.  Looks good.

**Comment by mwtoews** [comment]
If we are requiring `shapely>=2.0`, then update `pyproject.toml` and `etc/environment.yml`.

**Comment by dbrakenhoff** [comment]
> If we are requiring `shapely>=2.0`, then update `pyproject.toml` and `etc/environment.yml`.

Good spot. Done!

**Pull request merged**